spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/sql/TestSparkTruncateFunction.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.sql;
+
+import java.io.UnsupportedEncodingException;
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.relocated.com.google.common.io.BaseEncoding;
+import org.apache.iceberg.spark.SparkCatalogTestBase;
+import org.apache.spark.sql.AnalysisException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestSparkTruncateFunction extends SparkCatalogTestBase {
+  private static final Namespace NS = Namespace.of("db");
+
+  private final String fullNamespace;
+
+  public TestSparkTruncateFunction(String catalogName, String implementation, Map<String, String> config) {
+    super(catalogName, implementation, config);
+    this.fullNamespace = ("spark_catalog".equals(catalogName) ? "" : catalogName + ".") + NS;
+  }
+
+  @Before
+  public void createNamespace() {
+    sql("CREATE NAMESPACE IF NOT EXISTS %s", fullNamespace);
+    sql("USE %s", fullNamespace);
```

Comment:
I don't think you need a namespace if you always reference `system.truncate`.

Reply:
That's correct. I'm still working on how to handle the `system` namespace.

I think probably we just register it as a special namespace within the `SparkSessionCatalog`? The analyzer specifically checks if the namspace exists when using the session catalog.

Reply:
Removed the namespace from here as it's not needed.

As mentioned here https://github.com/apache/iceberg/pull/5305#discussion_r927178316, we will hold off on `SparkSessionCatalog` and then follow up with handling the need for `system` namespace in spark session catalog in another PR.